pkg/runtest/run_test.go

```diff
@@ -278,7 +278,7 @@ func testCover(t *testing.T, target *prog.Target) {
 		// 32-bit comparisons must be the same, so test only a subset.
 		{
 			Is64Bit: 0,
-			Input: makeComps(
+			Input: makeComps32(
```

Comment:
Kernel always writes them as 64-bit values:
https://elixir.bootlin.com/linux/v6.10-rc3/source/kernel/kcov.c#L232
so the single version for makeComps was intentional.
Syz-executor regardless of own and kernel bitness should parse them as uint64. So of the test fails, there is something to fix in the executor instead.

Reply:
```
static void cover_collect(cover_t* cov)
{
	if (is_kernel_64_bit)
		cov->size = *(uint64*)cov->data;
	else
		cov->size = *(uint32*)cov->data;
}
```

This seems to be incorrect for comps in 32-bit mode.
`is_kernel_64_bit` is set to `false` in `syz_inject_cover()` but size is 64 bits for comps.

Reply:
That's coverage (just PCs), comparisons are read by this code:
https://github.com/google/syzkaller/blob/master/executor/executor.cc#L1121

Reply:
yes, but
`uint32 ncomps = th->cov.size;` and `th->cov.size` is 0 because of `cover_collect()`.

Reply:
Ah, right.
Kernel writes uint64 there, right?
https://elixir.bootlin.com/linux/v6.10-rc3/source/kernel/kcov.c#L255

So it's a bug in cover_collect. I guess comparisons were never working for s390x then, now thanks to the new test, we found this.

Reply:
maybe don't read size from `th->cov.size` but use
```
uint64 ncomps = *(uint64_t*)(th->cov.data;
```

Reply:
No, scratch that.
It's still a bug in executor, but only in executor_test.g code. executor_linux.h is doing the right thing, so it worked.

Reply:
> uint64 ncomps = *(uint64_t*)th->cov.data;

Yes, you are right.

Scratch all that said recently :)